### PR TITLE
Infer model type from config

### PR DESCRIPTION
**Pull request by bogdankostic**
Until now, model types are inferred from the model name, which can cause problems like in #506. This PR makes it possible to infer the model type from the config files.

### CODE REVIEW

File: farm/modeling/language_model.py [hunk 1]
```diff
@@ -162,30 +162,68 @@ def load(cls, pretrained_model_name_or_path, n_added_tokens=0, language_model_cl
...
+            language_model_class = "Bert"
+        elif model_type == "xlnet":
+            language_model_class = "XLNet"
+        elif model_type == "electra":
+            language_model_class = "Electra"
+        elif model_type == "dpr":
+            if config.architectures[0] == "DPRQuestionEncoder":
+                language_model_class = "DPRQuestionEncoder"
+            elif config.architectures[0] == "DPRContextEncoder":
+                language_model_class = "DPRContextEncoder"
```
Comment (Timoeller) [inline farm/modeling/language_model.py:194]: What abou dpr reader? 
Should we also fall back to string matching if the architecture doesnt match?

File: farm/modeling/tokenization.py [hunk 2]
```diff
@@ -175,6 +141,97 @@ def load(cls, pretrained_model_name_or_path, tokenizer_class=None, use_fast=Fals
...
+            tokenizer_class = "BertTokenizer"
+        elif model_type == "xlnet":
+            tokenizer_class = "XLNetTokenizer"
+        elif model_type == "electra":
+            tokenizer_class = "ElectraTokenizer"
+        elif model_type == "dpr":
+            if config.architectures[0] == "DPRQuestionEncoder":
+                tokenizer_class = "DPRQuestionEncoderTokenizer"
+            elif config.architectures[0] == "DPRContextEncoder":
+                tokenizer_class = "DPRContextEncoderTokenizer"
```
Comment (Timoeller) [inline farm/modeling/tokenization.py:181]: dpr reader +
fall back to string matching here as well?

**Review by Timoeller — COMMENTED** [review]
Looking already good.

I made a suggestion, though it might not possible to include dprReader for now (we are waiting for the transformers implementation)

File: farm/modeling/language_model.py (hunk 1 above)
Comment (bogdankostic) [reply farm/modeling/language_model.py:194]: DPR Reader is not supported at the moment (see https://github.com/deepset-ai/haystack/issues/440). Should we add a NotImplementedError?

**Review by Timoeller — APPROVED** [review]
Looking good

**Pull request merged**